pkg/manifests/config.go

```diff
@@ -473,6 +482,46 @@ func (c *Config) NoProxy() string {
 	return c.ClusterMonitoringConfiguration.HTTPConfig.NoProxy
 }
 
+// PodCapacityReader returns the maximum number of pods that can be scheduled in a cluster.
+type PodCapacityReader interface {
+	PodCapacity(context.Context) (int, error)
+}
+
+func (c *Config) LoadEnforcedBodySizeLimit(pcr PodCapacityReader, ctx context.Context) error {
+	if c.ClusterMonitoringConfiguration.PrometheusK8sConfig.EnforcedBodySizeLimit == "automatic" {
+		podCapacity, err := pcr.PodCapacity(ctx)
+		if err != nil {
+			return fmt.Errorf("error fetching pod capacity: %v", err)
+		}
+		c.ClusterMonitoringConfiguration.PrometheusK8sConfig.EnforcedBodySizeLimit = calculateBodySizeLimit(podCapacity)
+		return nil
+	}
+
+	if c.ClusterMonitoringConfiguration.PrometheusK8sConfig.EnforcedBodySizeLimit != "" {
+		return poperator.ValidateSizeField(c.ClusterMonitoringConfiguration.PrometheusK8sConfig.EnforcedBodySizeLimit)
+	}
+
+	return nil
+}
+
+func (c *Config) UseMinimalEnforcedBodySizeLimit() {
+	c.ClusterMonitoringConfiguration.PrometheusK8sConfig.EnforcedBodySizeLimit = fmt.Sprintf("%dMB", int(math.Ceil(float64(minimalSizeLimit)/(1024*1024))))
+}
+
+func calculateBodySizeLimit(podCapacity int) string {
+	const samplesPerPod = 400       // 400 samples per pod
+	const sizePerSample = 200       // 200 Bytes
+	const loadFactorPercentage = 60 // assume 80% of the maximum pods capacity per node is used
+
+	bodySize := loadFactorPercentage * podCapacity / 100 * samplesPerPod * sizePerSample
+	if bodySize < minimalSizeLimit {
+		bodySize = minimalSizeLimit
+		klog.Infof("Calculated scrape body size limit is too small, using default value %v instead", minimalSizeLimit)
```

Comment:
we could log both values (e.g. "calculated body size limit = ... is too small, using ... instead")